### ISSUE TITLE
Get render process information from ICoreWebView2 object

### ISSUE THREAD

**Issue by ztextplus** [MicrosoftEdge/WebView2Feedback#3298]
**Is your feature request related to a problem? Please describe.**
We have multiple features that use the browser function, and now we really want to have such a requirement. 

We want to count the CPU and Memory usage of each feature, that is, **count the CPU and Memory usage of a certain feature for the Render process**. So I hope that Webview2 can provide an interface to get the corresponding set of Render process IDs from the  ICoreWebView2 object.
If this interface is provided, the usage process is as follows:

1. Get feature1's ICoreWebView2 object.
2. Call API ` GetRenderProsses `provided by ICoreWebView2 object.
3. Calculate the memory usage of the process returned by `GetRenderProsses`.

Although Webview2 provides the `GetProcessInfos`  interface, this interface obtains all the process information related to Webview2, we still cannot know the **Render process information corresponding to an ICoreWebView2**.

**Comment by aluhrs13** [comment]
@ztextplus - `CoreWebView2ProcessKind` is included in the result of `CoreWebView2ProcessInfo` for this purpose.

**Comment by novac42** [comment]
Hi @ztextplus , does andy's reply above solve the problem? If yes can the issue be closed?

**Comment by ztextplus** [comment]
@aluhrs13 @novac42  
Sorry, I don't think my question was answered.
For example:

Calling GetProcessInfos I will get an array of processes with two render processes (37696 and 21020), now the problem is:
how could I know:
37696 corresponds to the render process of featrue1
21020 corresponds to the render process of featrue2
or
37696 corresponds to the render process of featrue2
21020 corresponds to the render process of featrue1

Why is there such a requirement? We want to count the memory and CPU usage of featrue1 and featrue2 respectively.
If this interface is provided:
I can obtain the render process of featrue1 through the ICoreWebView2 object of featrue1, and the render process of featrue2 through the ICoreWebView2 object of featrue2, and then count their memory usage respectively.

**Comment by aluhrs13** [comment]
@ztextplus can you check out #3234? That is in-progress and might fit your need.

**Comment by JosephJin0815** [comment]
Hi, @ztextplus  can you explain more about feature1/feature2? Are they created from` iframes`? Like 
```
[main frame 
        [child_iframe1: feature1], 
        [child_iframe2: feature2],
        [child_frame...:]
 ]```

FYI, `IFrameMemoryUsage` API provides method to get renderer process id for each iframe/main frame.

**Comment by ztextplus** [comment]
@aluhrs13 @novac42  @JosephJin0815 
As shown in the figure below, I have two features, which are feature1 and feature2, and created two browsers to correspond to these two features. But in the process manager on the right, we can only know that **these two features have created four render processes**. Now the question is how to confirm:

**Among the four render processes, which are created by the browser corresponding to feature 1 and which are created by the browser corresponding to feature 2**

Using GetProcessInfos, you can get all process IDs, but you cannot get the corresponding relationship between feature1 and Render processes.

At present, the way I can think of is to kill a render. At this time, if feature1 does not display normally, I know that the render I just killed corresponds to feature1.

![image](https://user-images.githubusercontent.com/48123998/225231923-bbb4f03f-05d3-4599-a16b-9ecb835bdb1a.png)

**Comment by JosephJin0815** [comment]
I don't think this API support your case. There're different cases one browser have many renderer process. This api will support only iframe case, which you created iframes inside that main page, and then you can get renderer process id for that main page(ICoreWebView2) and iframe(ICoreWebView2Frame).

**Comment by ztextplus** [comment]
@novac42  @JosephJin0815 @aluhrs13  
Is there any update, we really want this feature, we use multiple features in our app, and each feature corresponds to one or more render processes.

In actual use, we will find that the CPU and Memory usage of a certain render process is very high, and we really want to locate which feature this rendering process corresponds to. If this API can be provided, it will be very convenient for us to locate the problem.

**Comment by novac42** [comment]
@ztextplus Sorry for the wait. Right now we don't have this feature, I've added it to our backlog and will evaluate the priority in grooming.

**Comment by dfghj44444** [comment]
I have same issue, and the workroud mentioned above is for C#, how does it work on C++

**Comment by JosephJin0815** [comment]
Hi @ztextplus, 
You can check [GetProcessExtendedInfos](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environment13?view=webview2-1.0.2478.35#getprocessextendedinfos)(C++) API, which provides the mapping between renderer process Id and associated frame Info.  

And each `ICoreWebView2`/`ICoreWebView2Frame` has the `FrameId` property now, so you should at least able to know the corresponding two renderer for the ICoreWebView2 object you created for each feature.

Also a reference about [Process model for WebView2 apps](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model?tabs=csharp)

**Comment by JosephJin0815** [comment]
@dfghj44444 Do you mean `CoreWebView2ProcessKind`? 
Here's the C++ API reference [ICoreWebView2ProcessInfo](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2processinfo?view=webview2-1.0.2535.41#get_kind)

**Comment by dfghj44444** [comment]
@JosephJin0815 In fact I want to get all render process ids(including the main frame and all sub-iframes) which belong to a web-page.

**Comment by JosephJin0815** [comment]
Hi @dfghj44444
The [GetProcessExtendedInfos](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environment13?view=webview2-1.0.2478.35#getprocessextendedinfos) API enables you to retrieve all renderer processes info with associated frame info including main frame and all sub-frames. To group sub-frames under their respective main frames, you can  use [ParentFrameInfo](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2frameinfo2?view=webview2-1.0.2478.35#get_parentframeinfo) property. In scenarios where there are multiple main frames, the [FrameId](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.frameid?view=webview2-dotnet-1.0.2478.35#microsoft-web-webview2-core-corewebview2-frameid) property is useful for identifying a specific main frame.
